File: core/src/main/java/cz/vhromada/common/validator/ValidationType.java
package cz.vhromada.common.validator;

/**
 * An enum represents type of validation.
 *
 * @author Vladimir Hromada
 */
public enum ValidationType {

    /**
     * New entity validation
     */
    NEW,

    /**
     * Update entity validation
     */
    UPDATE,

    /**
     * Existing entity validation
     */
    EXISTS,

    /**
     * Deep validation
     */
    DEEP,

    /**
     * Moving up entity validation
     */
    UP,

    /**
     * Moving down entity validation
     */
    DOWN

}

File: common-core/src/main/kotlin/com/github/vhromada/common/validator/ValidationType.kt
package com.github.vhromada.common.validator

/**
 * An enum represents type of validation.
 *
 * @author Vladimir Hromada
 */
enum class ValidationType {

    /**
     * New entity validation
     */
    NEW,

    /**
     * Update entity validation
     */
    UPDATE,

    /**
     * Existing entity validation
     */
    EXISTS,

    /**
     * Deep validation
     */
    DEEP,

    /**
     * Moving up entity validation
     */
    UP,

    /**
     * Moving down entity validation
     */
    DOWN

}
